client/config/config.go

```diff
@@ -89,15 +89,15 @@ func CreateClientConfig(ctx client.Context, customClientTemplate string, customC
 		}
 	}
 
-	conf, err := getClientConfig(configPath, ctx.Viper)
+	conf, err := getClientConfig(configPath)
 	if err != nil {
 		return ctx, fmt.Errorf("couldn't get client config: %w", err)
 	}
 
 	// we need to update KeyringDir field on client.Context first cause it is used in NewKeyringFromBackend
 	ctx = ctx.WithOutputFormat(conf.Output).
 		WithChainID(conf.ChainID).
-		WithKeyringDir(ctx.HomeDir).
+		WithKeyringDir(ctx.KeyringDir).
```

Comment:
We don't need this and I don't think it works.
https://github.com/cosmos/cosmos-sdk/blob/2a95a4abdd1e0ea99240861cae217dc7199804b1/client/cmd.go#L118

Have you checked, running `make init-simapp` and send transactions? In my recent trials it wasn't working.

Reply:
After making a few edits to the script so `init-simapp` works on nix, running the script directly seemed to work. It didn't produce any errors that I noticed.

Reply:
After rebasing onto #19060, transactions seem to work fine.

Reply:
Not sure how I feel about having `KeyringDir` set to `HomeDir` regardless of the `--keyring-dir` flag, but given you're correct about this being a no-op, I'll revert the change.